File: Data Structures & Algorithms/find-duplicate-integer/submission-0.py
class Solution:
    def findDuplicate(self, nums: List[int]) -> int:
        first = 0
        second = 1
        nums.sort()

        while second < len(nums):
            if nums[first] == nums[second]:
                return nums[first]
            first += 1
            second += 1
        
        return 0
        

File: Data Structures & Algorithms/remove-duplicates-from-sorted-array/submission-0.java
class Solution {
    public int removeDuplicates(int[] nums) {
        HashSet<Integer> numMap = new HashSet<>();
        int first = 0;
        int second = 0;
        int result = 0;

        while(second < nums.length){
            if (numMap.contains(Integer.valueOf(nums[first]))){
                if (numMap.contains(Integer.valueOf(nums[second]))){
                    second++;
                } else{
                    nums[first] = nums[second];
                    numMap.add(nums[first]);
                    first++;
                    second = first;
                    result++;
                }
            } else{
                numMap.add(nums[first]);
                first++;
                second++;
                result++;
            }
        }

        return result;
    }
}